NGit.Test/NGit.Api/ApplyCommandTest.cs

```diff
@@ -204,5 +204,29 @@ public virtual void TestModifyZ()
 				()[0]);
 			CheckFile(new FilePath(db.WorkTree, "Z"), b.GetString(0, b.Size(), false));
 		}
+
+        [Test]
+        public void ApplyingPatchWithInconsistentMacLineEndingsIsSuccessfullyApplied()
+        {
+            const string pre = "a\r\r\nb";
+            const string post = "a\r\nb";
+            var git = new Git(db);
+            var stream = new MemoryStream();
+            var modifiedFile = Path.Combine(db.WorkTree, "file.txt");
+
+            File.WriteAllText(modifiedFile, pre);
+            git.Add().AddFilepattern(".").Call();
+            git.Commit().SetMessage("Initial commit").Call();
+
+            File.WriteAllText(modifiedFile, post);
+            git.Diff().SetOutputStream(stream).Call();
+
+            stream.Position = 0;
+            File.WriteAllText(modifiedFile, pre);
+
+            git.Apply().SetPatch(stream).Call();
+
+            Assert.That(File.ReadAllText(modifiedFile), Is.EqualTo(post));
+        }
```

Comment:
I had a play with it and tried to conform the test with the others but it is apparent that this is a special thing and useCrlfFiles does not make sense for this test because it is explicitely testing an edge case. I am going to move this test out of this class.